mavros_extras/src/plugins/odom.cpp

```diff
@@ -198,35 +110,26 @@ class OdometryPlugin : public plugin::PluginBase {
 	 * @brief Handle ODOMETRY MAVlink message.
 	 *
 	 * Message specification: https://mavlink.io/en/messages/common.html#ODOMETRY
+	 *
+	 * Callback to odometry that is sent from FCU by mavlink_messages. It expects
+	 * the parent frame of the msg to be "local_origin_ned" and the child frame to be "fcu_frd".
+	 * The odometry message gets rotated such that parent frame and child frame are the desired
+	 * frames set in the rosparams "odometry/in/fcu_odom_*_id_des" (set in px4_config.yaml). If these frames are not part
+	 * of the usual tf tree, the user is supposed to add these frames to the existing tf tree.
+	 *
 	 * @param msg	Received Mavlink msg
-	 * @param att	ATTITUDE msg
+	 * @param att	ODOMETRY msg
 	 */
 	void handle_odom(const mavlink::mavlink_message_t *msg, mavlink::common::msg::ODOMETRY &odom_msg)
 	{
-		/*** Send fcu_frd <-> local_origin_ned transform ***/
-		geometry_msgs::TransformStamped transform;
-		transform.header.stamp = m_uas->synchronise_stamp(odom_msg.time_usec);
-		transform.header.frame_id = "local_origin_ned";
-		transform.child_frame_id = "fcu_frd";
-
-		tf::vectorEigenToMsg(Eigen::Vector3d(odom_msg.x, odom_msg.y, odom_msg.z), transform.transform.translation);
-		tf::quaternionEigenToMsg(ftf::mavlink_to_quaternion(odom_msg.q), transform.transform.rotation);
-
-		m_uas->tf2_broadcaster.sendTransform(transform);
-		/***************************************************/
-
 		/**
-		 * Required affine rotations to apply transforms
+		 * Required rotations to transform the FCU's odometry msg tto desired parent and child frame
 		 */
-		Eigen::Affine3d tf_parent2local;
-		Eigen::Affine3d tf_child2local;
-		Eigen::Affine3d tf_parent2body;
-		Eigen::Affine3d tf_child2body;
+		Eigen::Affine3d tf_parent2parentDes;
+		Eigen::Affine3d tf_child2childDes;
 
-		//! Lookup to the required trans
-		transform_lookup("local_origin_ned", "fcu_frd",
-			local_frame_orientation_in, body_frame_orientation_in_desired,
-			tf_parent2local, tf_child2local, tf_parent2body, tf_child2body);
+		transform_lookup( fcu_odom_parent_id_des, "map_ned", tf_parent2parentDes );
+		transform_lookup( fcu_odom_child_id_des,  "base_link_frd",          tf_child2childDes );
```

Comment:
Formatting is odd, please run the linter.

Reply:
@mhkabir  When using a new version of uncrustify I get the following error that is related with the uncrustify config:
mavros/tools/uncrustify-cpp.cfg:38 Unknown symbol 'align_number_left'
The inconvenient thing is that on Ubuntu 18.04 you have to install uncrustify from old binaries to get it to work.